### ISSUE TITLE
List reloads completely when changing the query

### ISSUE THREAD

**Issue by CiriousJoker** [gildaswise/firestore_ui#22]
Even if you're just appending `.limit(10)`, it reloads the full list and starts at scroll position 0.

**Comment by gildaswise** [comment]
The way the Firestore queries work make it so I can't build a proper scroll state because the items can change to literally anything else if you change the query. The support for changing queries is recent, and relies on having the query as a proper variable somewhere. If it's the same code as the previous issue, I've seen that you're building the query directly into the Widget, try changing that to a variable and using setState or something else, and notify the user that the whole screen is being changed. In my projects, I just add a quick progress indicator if the query changes.

**Comment by CiriousJoker** [comment]
Well I ended up writing a full stream manager that I can control to change the streams and cache them + manage limits for pseudo paging.
This stream then binds to a waterfall list (or whatever the pubdev package is called) and the 4 main tabs are in-memory for performance and the other potentially infinite are out of memory for efficiency.

I implemented the scroll behavior via a NotificationListener which updates the Streamcontroller.

Maybe a little over engineered but it works
